falcosidekick 2.28.0 (#1103)
falcosecurity/falco-website: content/en/blog/falcosidekick-2-28-0/index.md

---
title: Falcosidekick 2.28.0
linktitle: Falcosidekick 2.28.0
author: Thomas Labarussias
date: 2023-07-28
slug: falcosidekick-2-28-0
images:
  - /blog/falcosidekick-2-28-0/images/falcosidekick-featured.png
tags: ["Falcosidekick","Release"]
---

It's summer, it's hot, many people are on holiday, but the Falco community is still there. Six months after the previous version of Falcosidekick, version 2.28.0 is officially available.

The number of pulls of the official image from Dockerhub has also just reached 15M, which is just mind-blowing. It took 3 years to reach the first 5M pulls and now it's less than 6 months. Awesome!

![](images/falcosidekick-docker-pulls.png)
![](images/falcosidekick-docker-pulls-last-year.png)

> Once again, thanks to the hard work of the community, we introduced a bunch of new outputs and features, we hope will be useful. You can read the full changelog [here](https://github.com/falcosecurity/falcosidekick/releases/tag/2.28.0).

Reviewer: ```suggestion
Once more, Falcosidekick expands Falco's integrability with a lot of new outputs. That and the introduction of many new features has been possible thanks to the hard work of the community. You can find a comprehensive list of these in the [changelog](https://github.com/falcosecurity/falcosidekick/releases/tag/2.28.0).
```

## New outputs

> At every release, the number of integrations of Falcosidekick increases, we reached 58 available outputs now!!! For sure, the threshold of 60 will be crossed soon. 

Reviewer: ```suggestion
With every new release, the number of integrations of Falcosidekick increases. We have reached the number of 58 available outputs already, and we hope the threshold of 60 will be left behind very soon.
```

### Redis

> [Redis](https://redis.com) is well known in-memory database, used for years. It's now a possible output of Falcosidekick, thanks to [pandyamarut](https://github.com/pandyamarut).

Reviewer: ```suggestion
[Redis](https://redis.com) is a well-known in-memory database with many years of adoption on its path. It is now possible for Falcosidekick to use it as an output destination, thanks to the contributions of [pandyamarut](https://github.com/pandyamarut).
```

### Telegram

> [Telegram](https://telegram.org/), the instant messaging platform, is more and more used by companies for notifications, thanks to [zufardhiyaulhaq](https://github.com/zufardhiyaulhaq), it can receive Falco alerts too.

Reviewer: ```suggestion
[Telegram](https://telegram.org/), the instant messaging platform, is becoming more and more used by companies for notifications, and thanks to [zufardhiyaulhaq](https://github.com/zufardhiyaulhaq), it can receive Falco alerts too.
```

![](images/falcosidekick-telegram.png)

### N8N

> You want to extend the possibilities or just avoid to develop a script to react to Falco events? Here's come [n8n](https://n8n.io/).

Reviewer: ```suggestion
Do you want to extend the possibilities or just avoid developing a script to react to Falco events? Here comes [n8n](https://n8n.io/).
```

![](images/falcosidekick-n8n.png)

### Grafana OnCall

> We met some Grafana maintainers at last Kubecon and we discussed about integrating Falco, with Falcosidekick, in the OnCall project. It's done.

Reviewer: ```suggestion
At the last KubeCon, we met with some of the Grafana maintainers. We discussed the integration of Falco using Falcosidekick within the OnCall project. It's done now.
```

![](images/falcosidekick-grafana-oncall.png)

### OpenObserve

[OpenObserve](https://openobserve.ai/) is a young but promising full stack observability platform.

![](images/falcosidekick-openobserve.png)

## New features

> Aside new outputs, we introduced very important and useful new features. Let's recap them.

Reviewer: ```suggestion
Aside from new outputs, we introduced very important and useful new features. Let's do a recap of them.
```

### Use different methods for the Webhook output

> The Webhook output since its implementation used the http method `POST`, you now have choice between `POST` and `PUT`, extending the possible API Rest you can use with.

Reviewer: ```suggestion
Since its implementation, the Webhook output has only used the HTTP method `POST`. Now, you can choose between the `POST` and `PUT` methods to send your data, extending the catalog of possible REST APIs to use it with.
```

```yaml
webhook:
  method: "POST" # HTTP method: POST or PUT (default: POST)
```

### Replace the brackets in the payload

> Some Falco fields refer to lists and to reflect that their keys contain brackets, like `proc.args[0]`, `proc.args[1]`, etc. Unfortunatly, some outputs may refuse payloads with brackets in keys, this is why we introduced the possibilty to replace them with a chosen character:

Reviewer: ```suggestion
Some Falco fields refer to lists and reflect that their keys contain brackets, like `proc.args[0]`, `proc.args[1]`, etc. Unfortunately, some outputs may refuse payloads with brackets in keys. For this reason, we introduced the possibility of replacing them with any other chosen character:
```

```yaml
<!-- ``` -->

### Set custom headers for Loki, Elasticsearch and Grafana outputs

> If you protect your private instances of `Loki`, `Grafana` or `Elasticsearch` you may need to specify custom headers.

Reviewer: ```suggestion
If you want to protect your private instances of `Loki`, `Grafana`, or `Elasticsearch` you may need to specify custom headers. This new feature allows you to do so.
```

```yaml
elasticsearch:

>   customHeaders: # Custom headers to add in POST, useful for Authentication

Reviewer: ```suggestion
  customHeaders: # Custom headers to add in POST. Useful for Authentication.
```

    key: value
```

### Match the priority with the severity for the AlertManager output

> `AlertManager` is a pretty common systems at companies using `Prometheus`, til now, the mapping between the Priority of Falco events and the Severity of AlertManager was static, you can now define it as you wish thanks to [Lowaiz](https://github.com/Lowaiz).

Reviewer: ```suggestion
`AlertManager` is a pretty common software at companies also using `Prometheus`. Until now, the mapping between the Priority of Falco events and the Severity of AlertManager was already predefined. You can now define it depending on your needs thanks to [Lowaiz](https://github.com/Lowaiz).
```

```yaml
alertmanager:
  customseveritymap: "" # comma separated list of tuple composed of a ':' separated Falco priority and Alertmanager severity that is used to override the severity label associated to the priority level of falco event. Example: debug:value_1,critical:value2. Default mapping: emergency:critical,alert:critical,critical:critical,error:warning,warning:warning,notice:information,informational:information,debug:information. (default: "")
```

### Set thresholds for the dropped events for the AlertManager output

> Another contribution from [Lowaiz](https://github.com/Lowaiz), you can now configure the thresholds to drop the events.

Reviewer: ```suggestion
Another contribution from [Lowaiz](https://github.com/Lowaiz): You can now configure a set of thresholds to start dropping the events.
```

```yaml
alertmanager:
  # dropeventdefaultpriority: "" # default priority of dropped events, values are emergency|alert|critical|error|warning|notice|informational|debug (default: "critical")
  dropeventthresholds: # comma separated list of priority re-evaluation thresholds of dropped events composed of a ':' separated integer threshold and string priority. Example: `10000:critical, 100:warning, 1:informational` (default: `"10000:critical, 1000:critical, 100:critical, 10:warning, 1:warning"`)
```

### Better integration with TimescaleDB

> [hileef](https://github.com/hileef) has improved the integration with `TimescaleDB`, thanks to him.

Reviewer: ```suggestion
We thank [hileef](https://github.com/hileef) for improving the integration with `TimescaleDB`.
```

### User roleARN and externalID for the AWS outputs

> In some situation, you may want Falcosidekick to assume a role, possibly from another account, you can do it with the new settings `rolearn` and `externalid`:

Reviewer: ```suggestion
Under some situations, you may want Falcosidekick to assume a role, possibly from another account. You can do it with the new pair of settings `rolearn` and `externalid`:
```

```yaml
aws:
<!-- ``` -->

### Set the region for the PagerDuty output

> `PagerDuty` proposes now 2 different regions, you can select which one to use.

Reviewer: ```suggestion
Falcosidekick allows you to select between one of the two different regions' `PagerDuty` offers now.
```

```yaml
pagerduty:
  region: "us" # Pagerduty Region, can be 'us' or 'eu' (default: us)
```

### Allow TLS for the SMTP output

> It's now possible to use TLS to communicate with the `SMTP` server.

Reviewer: ```suggestion
It is now possible to communicate with an `SMTP` server using TLS.
```

```yaml
smtp:
  tls: false # Use TLS connection (true/false). Default: true
```

### Set attributes to GCP PubSUb messages

> `GCP PubSub` accepts attributes in its messages, you can specify yours, thanks to [annadorottya](https://github.com/annadorottya).

Reviewer: ```suggestion
`GCP PubSub` accepts attributes in its messages. You can specify yours, thanks to [annadorottya](https://github.com/annadorottya).
```

```yaml
gcp:
  customAttributes: # Custom attributes to add to the Pub/Sub messages
    key: value
```

### More options for TLS and mTLS

> These are the biggest changes of this release. To improve the security, Falcosidekick can now listen in HTTPS, with TLS. You can also be more specific with the keys and certificates for the mTLS for the outputs.

Reviewer: ```suggestion
These are the most relevant changes of this release. To improve security, Falcosidekick can now listen using HTTPS with TLS. You can also be more specific with the keys and certificates for the mTLS for the outputs.
```

```yaml
mutualtlsfilespath: "/etc/certs" # folder which will used to store client.crt, client.key and ca.crt files for mutual tls for outputs, will be deprecated in the future (default: "/etc/certs")
  # certfile: "/etc/certs/client/client.crt" # client certification file
  keyfile: "/etc/certs/client/client.key" # client key
  cacertfile: "/etc/certs/client/ca.crt" # for server certification
tlsserver:
  # certfile: "/etc/certs/server/server.crt" # server certification file
  keyfile: "/etc/certs/server/server.key" # server key
<!-- ``` -->

{{% pageinfo color="warning" %}}
The `mutualtlsfilespath` setting is kept for now for backward compatibility but it will be remove in future
{{% /pageinfo %}}

> In some edge cases, you may need some endpoints to listen in HTTP only, you can specify them and the associated port:

Reviewer: ```suggestion
In some edge cases, you may need some endpoints to listen in HTTP only. You can specifically define them together with the associated port:
```

```yaml
tlsserver:
  notlsport: 2810 # port to serve http server serving selected endpoints (default: 2810)
  notlspaths: # if not empty, a separate http server will be deployed for the specified endpoints
    - "/metrics"
    - "/healthz"
```

> Thanks to [annadorottya](https://github.com/annadorottya) for her impressive work on this topic.

Reviewer: ```suggestion
Thanks to [annadorottya](https://github.com/annadorottya) for her impressive work on this functionality.
```

### Autocreate the topic for the Kafka output

> When Falcosidekick doesn't detect the topic, it can create it automatically. It's not enabled by default.

Reviewer: ```suggestion
When Falcosidekick doesn't detect the topic, it can create it automatically. This feature is not enabled by default.
```

```yaml
kafka:
<!-- ``` -->

### Support multiple bootstrap servers and TLS for the Kafka output

> To get a better resiliency, you can now specify several bootstrap servers and even communicate with them with TLS, thanks to [ibice](https://github.com/ibice).

Reviewer: ```suggestion
To get better resiliency, you can now specify several bootstrap servers and even communicate with them with TLS, thanks to [ibice](https://github.com/ibice).
```

```yaml
kafka:
  # tls: false # Use TLS for the connections (default: false)
```

## Fixes

We're not going to go into detail about all the corrections made in this version - you can find the full list [here](https://github.com/falcosecurity/falcosidekick/releases/tag/2.28.0). 

> Thanks to all those who have reported issues and to those who have corrected them.

Reviewer: ```suggestion
Thanks to everyone who reported issues and to those who have corrected them.
```

The most important have been:

> - Fix breaking brackets in `AWS SNS` messages
> - Fix setting name for the table of `TimescaleDB` output (thanks to [alika](https://github.com/alika))
> - Fix cardinality issue with `Prometheus` labels
> - Fix panic when assert output fields which are nil
> - Fix URL generation for `Spyderbat` output (thanks to [bc-sb](https://github.com/bc-sb))
> - Fix nil values in `Spyderbat` output (thanks to [spider-guy](https://github.com/spider-guy))
> - Fix duplicated headers in `SMTP` output (thanks to [apsega](https://github.com/apsega))

Reviewer: ```suggestion
- Fix breaking brackets in `AWS SNS` messages.
- Fix setting name for the table of `TimescaleDB` output (thanks to [alika](https://github.com/alika)).
- Fix the cardinality issue with `Prometheus` labels.
- Fix panic when asserting output fields that are `nil`.
- Fix URL generation for `Spyderbat` output (thanks to [bc-sb](https://github.com/bc-sb)).
- Fix `nil` values in `Spyderbat` output (thanks to [spider-guy](https://github.com/spider-guy)).
- Fix duplicated headers in `SMTP` output (thanks to [apsega](https://github.com/apsega)).
```

## Conclusion

> The respective Helm charts will be up to date soon to allow you to test on your own all these great new features. When ready, run a `helm upgrade --reuse-values -n falco` to do so.

Reviewer: ```suggestion
The respective Helm charts will be updated soon to allow you to test for yourself all these great new features. When it becomes ready, just issue the `helm upgrade --reuse-values -n falco` command to do so.
```

> Falcosidekick is now mentionned in the official [Falco docs](https://falco.org/docs/alerts/forwarding/), it's a shy departure, more details will come in future.

Reviewer: ```suggestion
Falcosidekick is now mentioned in the official [Falco docs](https://falco.org/docs/alerts/forwarding/). It's a shy beginning, but more details will come shortly.
```

> Once again, thanks to all adopters and contributors who helped and contributed for years to this projet, we'll never reached that success without you all.

Reviewer: ```suggestion
Once again, thanks to all the adopters and contributors who helped and contributed to this project all these years. We would never have reached this success without you.
```
